server/interpreter.js

```diff
@@ -479,21 +469,19 @@ Interpreter.prototype.initObject = function(scope) {
       Interpreter.NONENUMERABLE_DESCRIPTOR);
 
   wrapper = function(obj, prop) {
-    prop = (prop || thisInterpreter.UNDEFINED).toString();
+    throwIfNullUndefined(obj);
+    prop += '';
```

Comment:
I guess we need to balance that against the fact that `+ ''` calls `valueOf` in preference to `toString`, as you have recently discovered.  That might explain the speed difference: producing the wrong answer might take less time / do less allocating...